apache/metron
# METRON-1622 Allow user to define global property 'threat.triage.score.field' in Ambari

NOTE: This PR was built on top of #1047 and so includes those commits which will be cleared up once that PR is merged.  The change is almost exactly the same as #1047, except for a different field.

## Description

Based on #1055 and #1062 , the user can specify the name of the field containing the threat triage score in the global configuration. This is used by the Alerts UI.  Currently a user can only change this value using the CLI. This property should be exposed to the user via Ambari. The user should be able to define this property directly in Ambari.

## Testing

1. Launch the development environment.  Ensure alerts are visible in the Alerts UI and that the Service Check passes.

1. Open the REPL and validate the current value of the global property 'threat.triage.score.field'.  The value here, the default value should be `threat:triage:score`.

    ```
    [root@node1 ~]# source /etc/default/metron
    [root@node1 ~]# $METRON_HOME/bin/stellar -z $ZOOKEEPER
    Stellar, Go!
    ...
    [Stellar]>>> globals := CONFIG_GET("GLOBAL")
    {
      ...
      "threat.triage.score.field" : "threat:triage:score",
      ...
    }
    ```

1. Change the value in Ambari by going to Metron > Configs > REST > Source Type Field Name.

    ![screen shot 2018-06-18 at 5 03 45 pm](https://user-images.githubusercontent.com/2475409/41562545-f3eadd76-731a-11e8-91a3-9abea3b4063b.png)


1. After saving the change, Ambari should prompt for the following services to be restarted.

    > 1 Metron Alerts UI, 1 Metron Management UI, 1 Metron REST

    ![screen shot 2018-06-18 at 2 34 57 pm](https://user-images.githubusercontent.com/2475409/41555150-cdcc43ca-7304-11e8-8348-f91696c945a0.png)

1. Restart all affected services.

1. After the services have been restarted, open the REPL and validate that the value of the global property 'threat.triage.score.field' changed in the global config.

    ```
    [Stellar]>>> globals := CONFIG_GET("GLOBAL")
    {
      ...
      "threat.triage.score.field" : "different:threat:triage:score:field",
      ...
    }
    ```

## Pull Request Checklist
